engine/graphics/src/vulkan/graphics_vulkan.cpp

```diff
@@ -3177,6 +3177,19 @@ namespace dmGraphics
         //        per mipmap instead of batching in one cmd
         if (useStageBuffer)
         {
+            uint32_t slice_size = texDataSize / layer_count;
+
+        #ifdef __MACH__
+            // Note: There is an annoying validation issue on osx for layered compressed data that causes a validation error
+            //       due to misalignment of the data when using a stage buffer. The offsets in the stage buffer needs to be
+            //       8 byte aligned but for compressed data that is not the case for the lowest mipmaps.
```

Comment:
So basically aligning the mipmap data on a 8 or 16 byte alignment when writing, and allocating an aligned memory block when reading?

Reply:
Yes, I tried that and it didn't look correct. I'll have to revisit it.. I'll make an issue about it